api/src/main/java/org/openmrs/Condition.java

```diff
@@ -136,6 +150,43 @@ public static Condition copy(Condition fromCondition, Condition toCondition) {
 		toCondition.setDateVoided(fromCondition.getDateVoided());
 		return toCondition;
 	}
+
+	/**
+	 * Compares properties with those in the given Condition to determine if they have the same meaning
+	 * This method will return true immediately following the creation of a Condition from another Condition
+	 * @see Condition#newInstance(Condition)
+	 * This method will return false if any value is different, excepting identity data (id, uuid)
+	 * If the given instance is null, this will return false
+	 * @param c the Condition to compare against 	
+	 * @return true if the given Condition has the same meaningful properties as the passed Condition
+	 * @since 2.6.1
+	 */
+	public boolean matches(Condition c) {
+		if (c == null) {
+			return false;
+		}
+		CodedOrFreeText coft1 = getCondition() == null ? new CodedOrFreeText() : getCondition();
+		CodedOrFreeText coft2 = c.getCondition() == null ? new CodedOrFreeText() : c.getCondition();
+		
+		boolean ret = (OpenmrsUtil.nullSafeEquals(getPreviousVersion(), c.getPreviousVersion()));
+		ret = ret && (OpenmrsUtil.nullSafeEquals(getPatient(), c.getPatient()));
+		ret = ret && (OpenmrsUtil.nullSafeEquals(getEncounter(), c.getEncounter()));
+		ret = ret && (OpenmrsUtil.nullSafeEquals(getFormNamespaceAndPath(), c.getFormNamespaceAndPath()));
+		ret = ret && (OpenmrsUtil.nullSafeEquals(getClinicalStatus(), c.getClinicalStatus()));
+		ret = ret && (OpenmrsUtil.nullSafeEquals(getVerificationStatus(), c.getVerificationStatus()));
+		ret = ret && (OpenmrsUtil.nullSafeEquals(coft1.getCoded(), coft2.getCoded()));
+		ret = ret && (OpenmrsUtil.nullSafeEquals(coft1.getSpecificName(), coft2.getSpecificName()));
+		ret = ret && (OpenmrsUtil.nullSafeEquals(coft1.getNonCoded(), coft2.getNonCoded()));
```

Comment:
OK, @ibacher .  I removed the changes that added in hashcode and equals methods to CodedOrFreeText, and am now just comparing each property directly here.